### ISSUE TITLE
Program crashes on start

### ISSUE THREAD

**Issue by ayoungethan** [sgpthomas/hourglass#203]
[ 2370.492839] show_signal_msg: 11 callbacks suppressed
[ 2370.492843] com.github.sgpt[10081]: segfault at 0 ip 00007f1b0c5a502b sp 00007fff7578aa00 error 4 in libc-2.33.so[7f1b0c589000+154000]
[ 2370.492851] Code: 40 08 89 d7 48 89 75 b8 85 c9 0f 85 c7 02 00 00 83 ff 01 0f 84 7e 01 00 00 83 ff 24 0f 87 75 01 00 00 48 8b 5d c0 49 8b 48 68 <48> 0f be 13 48 89 d0 f6 44 51 01 20 74 1a 0f 1f 80 00 00 00 00 48
[ 2374.445943] com.github.sgpt[10083]: segfault at 0 ip 00007f1175d4f02b sp 00007ffe1574d780 error 4 in libc-2.33.so[7f1175d33000+154000]
[ 2374.445962] Code: 40 08 89 d7 48 89 75 b8 85 c9 0f 85 c7 02 00 00 83 ff 01 0f 84 7e 01 00 00 83 ff 24 0f 87 75 01 00 00 48 8b 5d c0 49 8b 48 68 <48> 0f be 13 48 89 d0 f6 44 51 01 20 74 1a 0f 1f 80 00 00 00 00 48
[ 2415.858411] com.github.sgpt[10207]: segfault at 0 ip 00007fce9af6b02b sp 00007fff058de050 error 4 in libc-2.33.so[7fce9af4f000+154000]
[ 2415.858421] Code: 40 08 89 d7 48 89 75 b8 85 c9 0f 85 c7 02 00 00 83 ff 01 0f 84 7e 01 00 00 83 ff 24 0f 87 75 01 00 00 48 8b 5d c0 49 8b 48 68 <48> 0f be 13 48 89 d0 f6 44 51 01 20 74 1a 0f 1f 80 00 00 00 00 48
[ 2434.839159] com.github.sgpt[10209]: segfault at 0 ip 00007f81276eb02b sp 00007ffc0bdf3a20 error 4 in libc-2.33.so[7f81276cf000+154000]
[ 2434.839183] Code: 40 08 89 d7 48 89 75 b8 85 c9 0f 85 c7 02 00 00 83 ff 01 0f 84 7e 01 00 00 83 ff 24 0f 87 75 01 00 00 48 8b 5d c0 49 8b 48 68 <48> 0f be 13 48 89 d0 f6 44 51 01 20 74 1a 0f 1f 80 00 00 00 00 48

**Comment by ayoungethan** [comment]
$ uname -a
Linux NanoQubeeOS 5.13.0-30-generic #33~20.04.1-Ubuntu SMP Mon Feb 7 14:25:10 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

elementary OS 6.1 Jólnir

**Comment by ayoungethan** [comment]
```
$ sudo flatpak update --commit=3422ebc995eb1ba3c5247ef8311a49de0650f2f7766c501d6e0ce466f7378def com.github.sgpthomas.hourglass
Looking for updates…
error: com.github.sgpthomas.hourglass not installed
```

Even though it is installed, uninstall fails:
`Application state not set as installed in AppCenter for package: Hourglass`

Manual install:
```
$ sudo flatpak install appcenter com.github.sgpthomas.hourglass
Looking for matches…

com.github.sgpthomas.hourglass permissions:
    ipc     fallback-x11     wayland     x11     bus ownership [1]

    [1] com.github.sgpthomas.client, com.github.sgpthomas.hourglass


        ID                                       Branch           Op           Remote              Download
 1. [✓] com.github.sgpthomas.hourglass           stable           i            appcenter           1.0 kB / 210.3 kB

Installing… ████████████████████ 100%
Note that '/var/lib/flatpak/exports/share' is not in the search path
set by the XDG_DATA_HOME and XDG_DATA_DIRS
environment variables, so applications may not
be able to find it until you set them. The
directories currently searched are:

- /root/.local/share
        ID                                       Branch           Op           Remote              Download
 1. [✓] com.github.sgpthomas.hourglass           stable           i            appcenter           1.0 kB / 210.3 kB

Installation complete.
```

**Comment by ayoungethan** [comment]
```
flatpak run com.github.sgpthomas.hourglass --help

** (com.github.sgpthomas.hourglass-daemon:4): CRITICAL **: 12:15:12.858: HourglassServer.vala:61: Could not aquire service name
```

**Comment by ayoungethan** [comment]
Interestingly, I have a second user on this account and hourglass runs correctly in their account (non-admin, standard privileges).

Will only run with root privileges in the computer's admin user account:
`$ sudo -i
root@NanoQubeeOS:~# flatpak run com.github.sgpthomas.hourglass 

** (com.github.sgpthomas.hourglass-daemon:4): CRITICAL **: 12:21:39.014: HourglassServer.vala:61: Could not aquire service name
Gtk-Message: 12:21:39.031: Failed to load module "canberra-gtk-module"
Gtk-Message: 12:21:39.032: Failed to load module "canberra-gtk-module"

** (com.github.sgpthomas.hourglass:2): CRITICAL **: 12:21:39.037: file ../src/Services/DBusManager.vala: line 30: unexpected error: Error spawning command line “dbus-launch --autolaunch=e22a533e9bc9a5653b3b67bc6217dfaa --binary-syntax --close-stderr”: Child process exited with code 1 (g-spawn-exit-error-quark, 1)

(com.github.sgpthomas.hourglass:2): GLib-GObject-WARNING **: 12:21:39.048: invalid (NULL) pointer instance

(com.github.sgpthomas.hourglass:2): GLib-GObject-CRITICAL **: 12:21:39.048: g_signal_connect_object: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

** (com.github.sgpthomas.hourglass:2): CRITICAL **: 12:21:39.048: hourglass_services_hourglass_client_get_alarm_list: assertion 'self != NULL' failed

** (com.github.sgpthomas.hourglass:2): CRITICAL **: 12:21:39.048: hourglass_services_hourglass_client_get_alarm_list: assertion 'self != NULL' failed

** (com.github.sgpthomas.hourglass:2): CRITICAL **: 12:21:39.073: Settings.vala:95: Could not connect: No such file or directory

** (com.github.sgpthomas.hourglass:2): CRITICAL **: 12:21:39.073: Settings.vala:136: Could not connect: No such file or directory
`

**Comment by ryonakano** [comment]
Thanks for your detailed issue report! According to these logs, you seemed to install Hourglass as the root user, not as your user. I guess this is why your installation of Hourglass only runs with the root privileges. I'm not sure why the app has been installed in that way but maybe you installed it from root terminal? Anyway you should fix this issue with the following steps:

1. Open root terminal by `sudo -i`
2. Run `flatpak uninstall com.github.sgpthomas.hourglass`
3. Close the root terminal
4. Open AppCenter and search for `hourglass` and then install the app there

**Comment by ayoungethan** [comment]
Hello! I did install hourglass via appcenter, is the strange thing! And I tried uninstalling and reinstalling. I had installed it as a system wide installation via appcenter, and today it just stopped working for the admin user account.

I don't know what caused it, but I ended up doing a complete system reinstall, and it is now working again. But I'm not sure why it went from working to non-working overnight, with no known modifications to the system in-between (apart from normal system updates).
